### PR TITLE
angular/api: update ui asynchronously

### PR DESCRIPTION
**Pull request by proppy**
/cc @stephenplusplus @s3ththompson

I think the input[text] is not showing as disabled because of color: inherit in common.css: https://github.com/tastejs/todomvc-common/blob/master/base.css#L123

Demo: http://async.gcloud-todos.appspot.com/examples/angularjs

### CODE REVIEW

**Comment by addyosmani** [comment]
Where did we end up landing with this work?

**Comment by proppy** [comment]
I don't think it ever landed :/

File: examples/angularjs/js/services/todoStorage.js [hunk 1]
```diff
@@ -28,39 +28,35 @@ angular.module('todomvc')
...
-					}
-				});
-
-				angular.copy(incompleteTodos, store.todos);
+					angular.copy(incompleteTodos, store.todos);
+				}
 
 				return $http.delete('/api/todos')
 					.then(function success() {
+                                                update();
```
Comment (arthurvr) [inline examples/angularjs/js/services/todoStorage.js]: Also there's some wrong indentation going on in here.

File: examples/angularjs/js/services/todoStorage.js [hunk 2]
```diff
@@ -28,39 +28,35 @@ angular.module('todomvc')
...
 
 			clearCompleted: function () {
-				var originalTodos = store.todos.slice(0);
+				function update() {
+					var completeTodos = [], incompleteTodos = [];
+					store.todos.forEach(function (todo) {
+					    if (todo.completed) {
+						   completeTodos.push(todo);
+					    } else {
+						   incompleteTodos.push(todo);
```
Comment (arthurvr) [inline examples/angularjs/js/services/todoStorage.js]: wrong indentation. We use tabs.

**Comment by samccone** [comment]
So I am a bit confused as to why we want to handle the async path in this repo since we always use local storage on the angular todo site. 

Essentially this is a dead and untested code path for us, does anyone have thoughts on this?

For now I am :+1: to closing

**Comment by proppy** [comment]
It's not exactly dead ;) The backend demo linked on todomvc.com uses this

**Comment by samccone** [comment]
right, but it is not hosted by us as far as I know
http://gcloud-todos.appspot.com/examples/angularjs/#/

(unless it is) this code is not being used by us right?

**Comment by proppy** [comment]
I mean if it is linked from the front page, we might as well take care of it :)

I initially set up the hosting in coordination with @stephenplusplus and @addyosmani.

**Comment by urrgur** [comment]
merge conflict needed

**Comment by arthurvr** [comment]
This needs a rebase.

**Comment by samccone** [comment]
Going to close this, it has become stale. when someone rebases this we can reopen :dancer:

**Pull request closed without merging**